game-core/src/main/java/games/strategy/engine/lobby/client/login/LobbyServerPropertiesFetcher.java

```diff
@@ -56,6 +60,14 @@ public LobbyServerPropertiesFetcher() {
    *         Otherwise backup values from client config.
    */
   public Optional<LobbyServerProperties> fetchLobbyServerProperties() {
+    if (lobbyServerProperties == null) {
+      final Optional<LobbyServerProperties> props = fetchProperties();
+      props.ifPresent(lobbyProps -> lobbyServerProperties = lobbyProps);
+    }
+    return Optional.ofNullable(lobbyServerProperties);
+  }
+  
+  private Optional<LobbyServerProperties> fetchProperties() {
```

Comment:
Just noting that this method (and thus `fetchLobbyServerProperties()`) appears to always returns a non-empty `Optional`, so that its return type could be simplified to just `LobbyServerProperties`.

Reply:
Agree that `fetchProperties` never returns an empty optional, but the caller method that is not the case.


```
  public Optional<LobbyServerProperties> fetchLobbyServerProperties() {
    if (lobbyServerProperties == null) {
      final Optional<LobbyServerProperties> props = fetchProperties();
      props.ifPresent(lobbyProps -> lobbyServerProperties = lobbyProps);
    }
    return Optional.ofNullable(lobbyServerProperties);
  }
```

If `lobbyServerProperties == null` then we should skip down to the `Optional.ofNullable(lobbyServerProperties)`, where we would return an empty optional.

Reply:
Scratch last comment, problem spotted, `fetchProperties` should be capable to return an empty. The 'lastused' is not guaranteed to be set.